server/src/test/java/io/crate/planner/operators/EquiJoinDetectorTest.java

```diff
@@ -119,4 +119,25 @@ public void test_equality_expression_followed_by_case_expression() {
         Symbol joinCondition = sqlExpressions.asSymbol("t1.a = t1.a AND CASE 1 WHEN t1.a THEN false ELSE t2.b in (t2.b) END");
         assertThat(EquiJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition)).isFalse();
     }
+
+    @Test
+    public void equality_and_many_relations_in_boolean_join_condition_hash_join_not_possible() {
+        Symbol joinCondition = sqlExpressions.asSymbol("(t1.a >= 1) = ((t1.a = t1.a) AND (t2.b <= t2.b))");
+        assertThat(EquiJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition)).isFalse();
+    }
+
+    @Test
+    public void inequality_in_boolean_join_condition_hash_join_possible() {
+        // Compare column with column
+        Symbol joinCondition = sqlExpressions.asSymbol("(t1.a >= 1) = (t2.b <= t2.b)");
+        assertThat(EquiJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition)).isTrue();
+
+        // Compare column with constant
+        joinCondition = sqlExpressions.asSymbol("(t1.a >= 1) = (t2.b < 10)");
+        assertThat(EquiJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition)).isTrue();
+
+        // Compare with column AND compare with constant.
+        joinCondition = sqlExpressions.asSymbol("(t1.a >= 1) = (t2.b <= t2.b AND t2.b < 10)");
+        assertThat(EquiJoinDetector.isHashJoinPossible(JoinType.INNER, joinCondition)).isTrue();
+    }
```

Comment:
All those work on master.  Added to improve coverage (as initially I thought <= is the problem).

Reply:
+1